pow/src/main/java/tech/pegasys/teku/pow/FallbackAwareEth1Provider.java

```diff
@@ -120,27 +132,34 @@ public FallbackAwareEth1Provider(
     return run(eth1Provider -> eth1Provider.ethGetLogs(ethFilter));
   }
 
-  private <T> SafeFuture<T> run(final Function<Eth1Provider, SafeFuture<T>> task) {
-    return run(task, eth1ProviderSelector.getProviders().iterator(), new Eth1RequestException());
+  private <T> SafeFuture<T> run(final Function<MonitorableEth1Provider, SafeFuture<T>> task) {
+    return run(task, eth1ProviderSelector.getValidProviderIterator(), new Eth1RequestException());
   }
 
   private <T> SafeFuture<T> run(
-      final Function<Eth1Provider, SafeFuture<T>> task,
-      final Iterator<Eth1Provider> providers,
+      final Function<MonitorableEth1Provider, SafeFuture<T>> task,
+      final Eth1ProviderSelector.ValidEth1ProviderIterator providers,
       Eth1RequestException exceptionsContainer) {
+
     return SafeFuture.of(
-        () ->
-            task.apply(providers.next())
+        () -> {
+          Throwable[] previousExceptions = exceptionsContainer.getSuppressed();
+          Optional<MonitorableEth1Provider> nextAvailableProvider = providers.next();
+          if (nextAvailableProvider.isPresent()) {
+            return task.apply(nextAvailableProvider.get())
                 .exceptionallyCompose(
                     err -> {
+                      LOG.warn("Retrying with next eth1 endpoint", err);
```

Comment:
This `WARN` and the error below will wind up being excessively noisy when a provider is unavailable.  We probably need to keep them at debug level until we address #3855 which can take a more wholistic approach to it.

Reply:
this should not be the case since I'm skipping non-working endpoints until the next validation check.